classes/ThankYous/DataTables/FilterDataTableSource.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Claromentis\ThankYou\ThankYous\DataTables;
+
+use Claromentis\Core\DataTable\ColumnFilter;
+use Claromentis\Core\DataTable\Contract\DataSource;
+use Claromentis\Core\Widget\Sugre\SugreUtility;
+use Claromentis\ThankYou\Api\ThankYous;
+use Date;
+
+abstract class FilterDataTableSource implements DataSource
+{
+	protected $api;
+
+	protected $sugre_utility;
+
+	public function __construct(ThankYous $thank_you_api, SugreUtility $sugre_utility)
+	{
+		$this->api           = $thank_you_api;
+		$this->sugre_utility = $sugre_utility;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function Filters()
+	{
+		return [
+			'from_date' => new ColumnFilter('c.date1', 'str'),
+			'to_date'   => new ColumnFilter('c.date2', 'str'),
+			'erm'       => new ColumnFilter('c.erm', 'str'),
```

Comment:
I don't think the returned results for this actually do anything. I believe this data would be used when using the inbuilt filtering provided, rather than managing it itself in the `Data` method in `DataSource`. I'm going to have it just return an empty array to clarify.

Reply:
Fixed in fdf989deae6d63982bfa43e15f5becfe64af12b1